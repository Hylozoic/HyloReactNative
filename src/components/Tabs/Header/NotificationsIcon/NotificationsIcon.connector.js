import { connect } from 'react-redux'
import { get } from 'lodash/fp'
import getMe from 'store/selectors/getMe'

export function mapStateToProps (state, props) {
  const currentUser = getMe(state)
  const showBadge = get('newNotificationCount', currentUser)
  return {showBadge}
}

export default connect(mapStateToProps)
