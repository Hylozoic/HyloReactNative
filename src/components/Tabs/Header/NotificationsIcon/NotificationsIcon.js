import React from 'react'

import BadgedIcon from '../../../BadgedIcon'
import styles from '../Header.styles'

export default function NotificationsIcon ({ showBadge, showNotifications }) {
  return <BadgedIcon
    name='Notifications'
    onPress={showNotifications}
    showBadge={showBadge}
    style={styles.icon}
  />
}
