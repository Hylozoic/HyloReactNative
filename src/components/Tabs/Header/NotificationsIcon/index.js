import React from 'react'

import BadgedIcon from '../../../BadgedIcon'
import styles from '../Header.styles'

export default function NotificationsIcon (props) {
  return <BadgedIcon
    name='Notifications'
    style={styles.icon}
    onPress={props.showNotifications}
  />
}
