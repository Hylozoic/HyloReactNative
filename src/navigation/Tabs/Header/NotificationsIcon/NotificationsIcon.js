import React from 'react'

import BadgedIcon from 'components/BadgedIcon'
import styles from '../Header.styles'

export default function NotificationsIcon ({ showBadge, showNotifications }) {
  return (
    <BadgedIcon
      name='Notifications'
      action={showNotifications}
      showBadge={showBadge}
      style={styles.icon}
    />
  )
}
