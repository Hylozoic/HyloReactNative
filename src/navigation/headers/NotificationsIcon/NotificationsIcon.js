import React from 'react'
import BadgedIcon from 'components/BadgedIcon'
import styles from '../header.styles'

export default function NotificationsIcon ({ showBadge, showNotifications }) {
  return (
    <BadgedIcon
      name='Notifications'
      action={showNotifications}
      showBadge={showBadge}
      style={styles.headerIcon}
    />
  )
}
