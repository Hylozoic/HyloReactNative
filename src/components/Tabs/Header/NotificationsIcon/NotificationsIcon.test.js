import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import NotificationsIcon from './index'

it('matches the last snapshot', () => {
  const renderer = new ReactShallowRenderer()
  renderer.render(<NotificationsIcon showNotifications={() => {}} />)
  const actual = renderer.getRenderOutput()
  expect(actual).toMatchSnapshot()
})
