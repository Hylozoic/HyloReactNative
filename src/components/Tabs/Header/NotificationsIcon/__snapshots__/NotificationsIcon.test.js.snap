// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the last snapshot 1`] = `
<BadgedIcon
  name="Notifications"
  onPress={[Function]}
/>
`;
